src/main/java/com/amazonaws/encryptionsdk/keyrings/KeyringTraceFlag.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except
+ * in compliance with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.amazonaws.encryptionsdk.keyrings;
+
+/**
+ * Enum representing the possible actions a keyring may take on the
+ * different wrapping keys it manages.
+ */
+public enum KeyringTraceFlag {
+
+    /**
+     * A flag to represent that a keyring has generated a plaintext data key.
+     */
+    WRAPPING_KEY_GENERATED_DATA_KEY,
+
+    /**
+     * A flag to represent that a keyring has created an encrypted data key.
+     */
+    WRAPPING_KEY_ENCRYPTED_DATA_KEY,
+
+    /**
+     * A flag to represent that a keyring has obtained the
+     * corresponding plaintext data key from an encrypted data key.
+     */
+    WRAPPING_KEY_DECRYPTED_DATA_KEY,
+
+    /**
+     * A flag to represent that the keyring has cryptographically
+     * bound the encryption context to a newly created encrypted data key.
+     */
+    WRAPPING_KEY_SIGNED_ENC_CTX,
+
+    /**
+     * A flag to represent that the keyring has verified that an encrypted
+     * data key was originally created with a particular encryption context.
+     */
+    WRAPPING_KEY_VERIFIED_ENC_CTX
+}
```

Comment:
Thoughts on changing these names to spell out "encryption context"? I know these are the magic strings that we've used to date, but since we decided we're not worrying about communicating them across implementations I kind of want to go back in the Python implementation and expand this.

Reply:
Sounds good to me. And actually the spec already spells out the word anyway (https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/structures.md#flags). I'll get rid of the "WRAPPING_KEY_" part too, I don't think that is adding much